### ISSUE TITLE
NS4D question:BC velocity loss missing

### ISSUE THREAD

**Issue by wangguan1995** [stnamjef/SPINN#5]
Hi there, 

When I read your code of NS4D, boundary loss made me confused, I will appreciate if you can help me:
  
Reference in Here : https://github.com/stnamjef/SPINN/blob/main/navier_stokes4d.py 

Line 91
``` python
def boundary_loss(params, t, x, y, z, w):
        loss = 0.
        for i in range(6):
            wx = vorx(apply_fn, params, t[i], x[i], y[i], z[i])
            wy = vory(apply_fn, params, t[i], x[i], y[i], z[i])
            wz = vorz(apply_fn, params, t[i], x[i], y[i], z[i])
            loss += (1/6.) * jnp.mean((wx - w[i][0])**2) + jnp.mean((wy - w[i][1])**2) + jnp.mean((wz - w[i][2])**2)
        return loss
```

Here for the boundary condition, the **velocity loss** term seems missing.
As a fair comparison, velocity loss is properly included in the Initial loss:

``` python
def initial_loss(params, t, x, y, z, w, u):
        ux, uy, uz = apply_fn(params, t, x, y, z)
        wx = vorx(apply_fn, params, t, x, y, z)
        wy = vory(apply_fn, params, t, x, y, z)
        wz = vorz(apply_fn, params, t, x, y, z)
        loss = jnp.mean((wx - w[0])**2) + jnp.mean((wy - w[1])**2) + jnp.mean((wz - w[2])**2)
        loss += jnp.mean((ux - u[0])**2) + jnp.mean((uy - u[1])**2) + jnp.mean((uz - u[2])**2)
        return loss
```

Need your ideas~ Thanks!

**Comment by jwcho5576** [comment]
Sorry for the late reply. Based on our experience, giving velocity ($u$) supervision was NOT mandatory throughout all of the loss functions (i.e., boundary/initial loss). Only using the vorticity ($\omega$) was enough. We gave velocity to the initial loss just because it showed a better performance in accuracy. Giving more constraints on the initial condition is known to be a better way of training PINNs for time-dependent PDEs. We did not use the velocity in the boundary loss, but might worth a try.
